pkg/controller/serviceaccount/legacy_serviceaccount_token_cleaner.go

```diff
@@ -183,7 +186,29 @@ func (tc *LegacySATokenCleaner) evaluateSATokens(ctx context.Context) {
 			continue
 		}
 
-		logger.Info("Delete auto-generated service account token", "secret", klog.KRef(secret.Namespace, secret.Name), "creationTime", secret.CreationTimestamp, "lastUsed", lastUsed)
+		invalidSince := secret.Labels[InvalidSinceLabelKey]
+		// If the secret has not been labeled with invalid since date or the label value has invalid format, update the invalidSince label with the current date value.
+		_, err = time.Parse(dateFormat, invalidSince)
+		if err != nil {
+			invalidSince = now.Format(dateFormat)
+			logger.Info("Mark the auto-generated service account token as invalid", "invalidSince", invalidSince, "secret", klog.KRef(secret.Namespace, secret.Name))
+			patchContent, err := json.Marshal(applyv1.Secret(secret.Name, secret.Namespace).WithLabels(map[string]string{InvalidSinceLabelKey: invalidSince}))
```

Comment:
and WithUID to make sure we only patch the secret instance we expect

Reply:
Added WithUID().